pymc/sampling.py

```diff
@@ -1620,10 +1620,20 @@ def sample_posterior_predictive(
 
     _trace: Union[MultiTrace, PointList]
     nchain: int
+    if idata_kwargs is None:
+        idata_kwargs = {}
+    else:
+        idata_kwargs = idata_kwargs.copy()
+    if "coords" not in idata_kwargs:
+        idata_kwargs["coords"] = {}
     if isinstance(trace, InferenceData):
+        idata_kwargs["coords"].setdefault("draw", trace.posterior.draw)
+        idata_kwargs["coords"].setdefault("chain", trace.posterior.chain)
```

Comment:
Mypy isn't broken - you just introduced two lines that according to the type information from ArviZ don't work.

Replace `idata.posterior` with `idata["posterior"]` in the two lines above

Reply:
Oohh, sorry. Now I see what Ravin meant in slack. These two lines prompted over 350 lines of mypy warnings and errors that start with aesaraf related things